backend/shared/shared/models/abstract_models.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from django.db import models
+from simple_history.models import HistoricalRecords
+
+
+class UUIDModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class HistoricalModel(models.Model):
+    """
+    Keeps a simple history of the changes for the model.
+    """
+
+    history = HistoricalRecords(inherit=True)
+
+    class Meta:
+        abstract = True
+
+
+class AbstractCompany(models.Model):
+    """
+    TODO: Shared company fields go here
+    """
+
+    class Meta:
+        abstract = True
```

Comment:
The TimeStampedModel could be shared with Kesäseteli as well, I'd like to use it with Application model

```
class TimeStampedModel(models.Model):
    created_at = models.DateTimeField(auto_now_add=True, verbose_name=_("time created"))
    modified_at = models.DateTimeField(auto_now=True, verbose_name=_("time modified"))

    class Meta:
        abstract = True

```

Reply:
Sure, I'll add it